package rekisterit;
// Generated by ComTest BEGIN
import rekisterit.*;
import static org.junit.Assert.*;
import org.junit.*;
// Generated by ComTest END

import tiedot.Tapahtuma;

/**
 * Test class made by ComTest
 * @version 2012.04.15 20:48:52 // Generated by ComTest
 *
 */
public class TapahtumatTest {



  // Generated by ComTest BEGIN
  /** 
   * testLisaa58 
   * @throws OmaException when error
   */
  @Test
  public void testLisaa58() throws OmaException {    // Tapahtumat: 58
    Tapahtumat Tapahtumat = new Tapahtumat(); 
    Tapahtuma Tapahtuma1 = new Tapahtuma(); 
    Tapahtuma Tapahtuma2 = new Tapahtuma(); 
    Tapahtuma Tapahtuma3 = new Tapahtuma(); 
    assertEquals("From: Tapahtumat line: 65", 0, Tapahtumat.getLukumaara()); 
    Tapahtuma1.rekisteroi(0); 
    Tapahtuma2.rekisteroi(1); 
    Tapahtuma1.vastaaTapahtuma(); 
    Tapahtuma2.vastaaTapahtuma(); 
    Tapahtumat.lisaa(Tapahtuma1); assertEquals("From: Tapahtumat line: 70", 1, Tapahtumat.getLukumaara()); 
    Tapahtumat.lisaa(Tapahtuma2); assertEquals("From: Tapahtumat line: 71", 2, Tapahtumat.getLukumaara()); 
    try {
    Tapahtumat.lisaa(Tapahtuma1); 
    fail("Tapahtumat: 72 Did not throw OmaException");
    } catch(OmaException _e_){ _e_.getMessage(); }
    assertEquals("From: Tapahtumat line: 73", Tapahtuma1, Tapahtumat.anna(1)); 
    assertEquals("From: Tapahtumat line: 74", Tapahtuma2, Tapahtumat.anna(2)); 
  } // Generated by ComTest END
}